### PR TITLE
Polishing Header component

### PR DESCRIPTION
**Pull request by saifulislamdev**
More meaningful commit messages :)

### CODE REVIEW

File: src/components/header.js [hunk 1]
```diff
@@ -9,25 +9,30 @@ import {
...
   const [expanded, setExpanded] = useState(false);
   const onClick = () => setExpanded((prvExpanded) => !prvExpanded);
 
-  const [isOpen1, setIsOpen1] = useState([false]);
-  const [isOpen2, setIsOpen2] = useState([false]);
+  /* nav dropdown expansions */
+  const [isOpen1, setIsOpen1] = useState(false);
+  const [isOpen2, setIsOpen2] = useState(false);
 
+  /* first dropdown items */
```
Comment (aannyyaa) [inline src/components/header.js:37]: Thanks for adding these comments highlighting what these are for!

File: src/components/header.js [hunk 2]
```diff
@@ -99,11 +108,13 @@ const Header = () => {
 
   return (
     <HeaderUSWDS extended={true}>
-      {/* <a class="usa-skipnav" href="#main-content">Skip to main content</a> */}
+      <a class="usa-skipnav" href="#main-content">
```
Comment (aannyyaa) [inline src/components/header.js]: I think there might be a way to do this through gatsby's Link component

File: src/styles/Header.css [hunk 3]
```diff
@@ -0,0 +1,3 @@
+.search-bar {
```
Comment (aannyyaa) [inline src/styles/Header.css]: When we have the styling files set up, this should probably be in our scss file of class names. I think we should also be using BEM

File: src/components/LanguageSwitcher.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import React from "react";
 import { Dropdown, Label } from "@trussworks/react-uswds";
 import { useTranslation } from "react-i18next";
-import "./LanguageSwitcher.css";
+import "../styles/LanguageSwitcher.css";
```
Comment (aannyyaa) [inline src/components/LanguageSwitcher.js:4]: I think we want to use scss files, not css

File: src/components/header.js (hunk 2 above)
Comment (saifulislamdev) [reply src/components/header.js]: They have something even better with `import { SkipNavLink, SkipNavContent } from "@reach/skip-nav";`. Check out the new push.

File: src/styles/Header.css (hunk 3 above)
Comment (saifulislamdev) [reply src/styles/Header.css]: I did this with the intention that I was doing it in BEM, but now I realize I wasn't. Nice catch. So would it be `.header__search`?

File: src/components/LanguageSwitcher.js (hunk 4 above)
Comment (saifulislamdev) [reply src/components/LanguageSwitcher.js:4]: In the last PR, I mentioned: 

> I believe Justin said a while back to use CSS files for custom classes.

 We can address this with him if you'd like.

Comment (aannyyaa) [reply src/components/LanguageSwitcher.js:4]: Yes, I believe we would have all our themes in the settings.scss file and then we'd have one or more scss files with class names. We want to use scss so we can access variables. I don't think we'd want to limit ourselves to a regular css file

File: src/components/header.js [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import React, { useState } from "react";
-import { Link, NavLink, BrowserRouter as Router } from "react-router-dom";
+import { Link, NavLink, BrowserRouter as Router } from "react-router-dom"; // TODO: use Link and NavLink from Gatsby and USWDS instead
```
Comment (aannyyaa) [inline src/components/header.js:2]: Might make sense to switch to gatsby Link in this PR

**Pull request merged**